src/android/com/megster/cordova/BluetoothSerial.kt

```diff
@@ -96,9 +92,29 @@ class BluetoothSerial : CordovaPlugin() {
         BluetoothSerialService.stop()
     }
 
-    private fun enableBluetoothIfNecessary() {
-        if (!bluetoothAdapter.isEnabled) {
-            bluetoothAdapter.enable()
+    private fun isEnabled(callbackContext: CallbackContext) {
+        try {
+            callbackContext.success(if (bluetoothAdapter.isEnabled()) 1 else 0)
+        } catch (e: Exception) {
+            Log.e(TAG, "Unable to check isEnabled: $e")
+            callbackContext.error(e.toString())
+        }
+    }
+
+    private fun enable(callbackContext: CallbackContext) {
+        try {
+            if (!bluetoothAdapter.isEnabled) {
+                // returns a boolean indicating whether it was able
+                // to begin enabling the adapter, which is an
+                // asynchronous process.
+                // https://github.com/aosp-mirror/platform_frameworks_base/blob/nougat-release/core/java/android/bluetooth/BluetoothAdapter.java#L899
+                callbackContext.success(if (bluetoothAdapter.enable()) 1 else 0)
```

Comment:
again - please don't write if statements like this - this is also dupe code - maybe it makes sens to break this out into a helper

Reply:
what exactly is dupe?

Reply:
`if (bluetoothAdapter.enable()) 1 else 0`

Reply:
the other function is working with isEnabled.  so they look similar but are not dupes.

Reply:
can we please break out the logic and not just chuck it into a method as a parameter - it is harder to read and follow